iis-wf/iis-wf-export-actionmanager/src/test/java/eu/dnetlib/iis/wf/export/actionmanager/relation/citation/Matchers.java

```diff
@@ -30,6 +35,32 @@ public void describeTo(Description description) {
             }
         };
     }
+    
+    private static boolean matchesCollectedFrom(List<KeyValue> source, List<KeyValue> target) {
+        if (source != null) {
+            if (target != null) {
+                if (source.size() == target.size()) {
+                    for (int i = 0; i < source.size(); i++) {
+                        if (!source.get(i).getKey().equals(target.get(i).getKey()) || 
+                                !source.get(i).getValue().equals(target.get(i).getValue())) {
+                            return false;
+                        }
+                    }
+                    return true;
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        } else {
+            if (target == null) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
```

Comment:
Way more branches than necessary, I think.
```suggestion
    private static boolean matchesCollectedFrom(List<KeyValue> source, List<KeyValue> target) {
        if (source != null && target != null && source.size() == target.size()) {
            for (int i = 0; i < source.size(); i++) {
                if (!source.get(i).getKey().equals(target.get(i).getKey()) || 
                        !source.get(i).getValue().equals(target.get(i).getValue())) {
                    return false;
                }
            }
            return true;
        }
        return source == null && target == null;
    }
```